src/query/functions-v2/src/scalars/variant.rs

```diff
@@ -43,4 +75,756 @@ pub fn register(registry: &mut FunctionRegistry) {
+                args_type: vec![DataType::Variant, DataType::Number(NumberDataType::UInt64)],
+                return_type: DataType::Nullable(Box::new(DataType::Variant)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_| None),
+            eval: Box::new(get_by_index_fn),
+        }))
+    });
+
+    registry.register_function_factory("get", |_, args_type| {
+        if args_type.len() != 2 {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "get",
+                args_type: vec![
+                    DataType::Nullable(Box::new(DataType::Variant)),
+                    DataType::Nullable(Box::new(DataType::Number(NumberDataType::UInt64))),
+                ],
+                return_type: DataType::Nullable(Box::new(DataType::Variant)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_| None),
+            eval: Box::new(wrap_nullable(get_by_index_fn)),
+        }))
+    });
+
+    registry.register_function_factory("get", |_, args_type| {
+        if args_type.len() != 2 {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "get",
+                args_type: vec![DataType::Variant, DataType::String],
+                return_type: DataType::Nullable(Box::new(DataType::Variant)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_| None),
+            eval: Box::new(get_by_name_fn),
+        }))
+    });
+
+    registry.register_function_factory("get", |_, args_type| {
+        if args_type.len() != 2 {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "get",
+                args_type: vec![
+                    DataType::Nullable(Box::new(DataType::Variant)),
+                    DataType::Nullable(Box::new(DataType::String)),
+                ],
+                return_type: DataType::Nullable(Box::new(DataType::Variant)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_| None),
+            eval: Box::new(wrap_nullable(get_by_name_fn)),
+        }))
+    });
+}
+
+fn get_by_index_fn(
+    args: &[ValueRef<AnyType>],
+    _: FunctionContext,
+) -> Result<Value<AnyType>, String> {
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+
+    let source_arg = args[0].try_downcast::<VariantType>().unwrap();
+    let index_arg = args[1].try_downcast::<UInt64Type>().unwrap();
+    let size = len.unwrap_or(1);
+    let mut bitmap = MutableBitmap::with_capacity(size);
+    let mut builder = StringColumnBuilder::with_capacity(size, 0);
+    for idx in 0..size {
```

Comment:
```suggestion
    for (s, index) in source_arg.iter().zip(index_arg.iter()) {
```

Reply:
`source_arg` and `index_arg` are `ValueRef`, don't have `iter()` method.